wolvic/browser/vr/wvr_manager.h

```diff
@@ -93,19 +88,31 @@ class WvrManager : public device::mojom::XRPresentationProvider,
       base::OnceCallback<void(device::mojom::XRSessionPtr)> callback);
   void CreateOrResizeWebXrSurface(const gfx::Size& size);
   void OnGpuProcessConnectionReady();
-  void CreateSurfaceBridge();
+  void CreateSurfaceBridge(gl::SurfaceTexture* surface_texture);
 
   void OnWebXrFrameAvailable();
   void OnWebXrTimedOut();
 
   std::vector<device::mojom::XRInputSourceStatePtr> GetInputSourceState();
 
-  bool CanStartNewAnimatingFrame();
-  void TryStartAnimatingFrame();
+  // Checks if we're in a valid state for starting animation of a new frame.
+  // Invalid states include a previous animating frame that's not complete
+  // yet (including deferred processing not having started yet), or timing
+  // heuristics indicating that it should be retried later.
+  bool WebVrCanAnimateFrame();
+  // Call this after state changes that could result in WebVrCanAnimateFrame
+  // becoming true.
+  void WebXrTryStartAnimatingFrame();
+
+  // Shared logic for SubmitFrame variants, including sanity checks.
+  // Returns true if OK to proceed.
+  bool SubmitFrameCommon(int16_t frame_index, base::TimeDelta time_waited);
```

Comment:
Can this method be moved to `private:` section?

Reply:
it was already within `private:`. no?

Reply:
You're right, my apologies.